File: frontend/app/src/main/java/com/example/studybuddy/retrofitNetwork/RetrofitClient.java
package com.example.studybuddy.retrofitNetwork;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

/**
 * Singleton wrapper around Retrofit for your StudyBuddyService.
 */
public class RetrofitClient {
    private static final String BASE_URL = "http://10.0.2.2:8081/"; // ktor backend server runs on port 8081
    private static RetrofitClient instance;
    private final StudyBuddyService service;

    private RetrofitClient() {
        Retrofit retrofit = new Retrofit.Builder()
                .baseUrl(BASE_URL)
                .addConverterFactory(GsonConverterFactory.create())
                .build();
        service = retrofit.create(StudyBuddyService.class);
    }

    // Returns Retrofit client instance
    public static synchronized RetrofitClient getInstance() {
        if (instance == null) {
            instance = new RetrofitClient();
        }
        return instance;
    }

    // Accesses API interface
    public StudyBuddyService getService() {
        return service;
    }
}

File: frontend/app/src/main/java/com/example/studybuddy/RetrofitClient.kt
package com.example.studybuddy

import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

object RetrofitClient {
    private const val BASE_URL = "https://api.openai.com/"

    private val loggingInterceptor = HttpLoggingInterceptor().apply {
        level = HttpLoggingInterceptor.Level.BODY // Or BASIC, HEADERS, NONE
    }

    private val okHttpClient = OkHttpClient.Builder()
        .addInterceptor(loggingInterceptor)
        .build()

    val instance: OpenAIService by lazy {
        val retrofit = Retrofit.Builder()
            .baseUrl(BASE_URL)
            .addConverterFactory(GsonConverterFactory.create())
            .client(okHttpClient)
            .build()

        retrofit.create(OpenAIService::class.java)
    }
}
